### ISSUE TITLE
Keep mesh renderer's material with multiple input registrations

### ISSUE THREAD

**Issue by StarChick971** [crest-ocean/crest#684]
When using more than one input registration for the same mesh, for instance _Register Flow Input_+ _Register Anim Waves input_ scripts, we have to use twice the same mesh to apply two different (shader) materials.

In my case, I have a mesh  I modify in run-time to create an animated wave. I use the same mesh twice times to :
- set water height with _Register Anim Wave Input_ script
- apply local forces with _Register Flow Input_ script
![image](https://user-images.githubusercontent.com/54045907/98467750-4ee13780-21d7-11eb-90bf-c71bff3e6048.png)
Plane mesh has **5151 vertices** here.

Would it be possible to keep usage of the same mesh, so, without the need to duplicate it to apply multiple input registrations?

**Comment by daleeidd** [comment]
Is the issue that you want to be able to share a mesh between two renderers, or have both of these inputs done in a single pass? We cannot do the latter since they are different render targets, but the former should be possible with scripting since we use Unity's mesh renderer.

**Comment by daleeidd** [comment]
I read your comment on discord and it is a bit more clear. A property on the register input component to set the mesh or mesh renderer. Is that correct?

**Comment by StarChick971** [comment]
> Is the issue that you want to be able to share a mesh between two renderers,
Yes, I think this is what I am trying to do using twice the same mesh, one for each renderer.

> I read your comment on discord and it is a bit more clear. A property on the register input component to set the mesh or mesh renderer. Is that correct?
Indeed within same _GameObject_ you have a _MeshFilter_ component which _mesh_ instance would be both used for:
->RegisterAnimInput
->RegisterFlowInput

Perhaps, this requires to duplicate the instantiated mesh for the second input?

**Comment by daleeidd** [comment]
@StarChick971 have you tried sharing a mesh using scripting as I have suggested? I think it may work for your case. You'll need two game objects, one for each input. Then assign the same mesh reference to each mesh renderer with a script.

**Comment by StarChick971** [comment]
@daleeidd I made a really simple script, which works well. It is getting TargetMeshFilter.mesh from my mesh GameObject then copy it to the MeshFilter of current GameObject with the register input.
Perhaps a "target MeshFilter" property on register input could be enough to add, which sets to current gameObject if not set by user?
I put down my code here:
`[RequireComponent(typeof(MeshFilter)), RequireComponent(typeof(MeshRenderer))]
public class MeshFilterDuplicate : MonoBehaviour
{
    public MeshFilter TargetFilter;
    private MeshFilter _currentFilter;

    void Start()
    {
        _currentFilter = GetComponent<MeshFilter>();
        if(TargetFilter != null)
        {
            if(TargetFilter.mesh != null)
                _currentFilter.mesh = TargetFilter.mesh;
        }
    }

    void Update()
    {
        if(TargetFilter != null)
        {
            if (TargetFilter.mesh != null)
            {
                _currentFilter.mesh = TargetFilter.mesh;
            }
        }
    }
}`

**Comment by daleeidd** [comment]
Maybe in the future if we move away from the mesh renderer and use _draw mesh_. But for now, your solution of working with the mesh renderer and mesh filter works. We don't change the mesh filter or mesh renderer so we aren't in the way of anything.

I'll close this but if you have anything further to add I will see your reply.
